src/Compilers/CSharp/Portable/Emitter/EditAndContinue/CSharpSymbolMatcher.cs

```diff
@@ -561,6 +561,57 @@ public override Symbol VisitPointerType(PointerTypeSymbol symbol)
                 return new PointerTypeSymbol(symbol.PointedAtTypeWithAnnotations.WithTypeAndModifiers(otherPointedAtType, otherModifiers));
             }
 
+            public override Symbol VisitFunctionPointerType(FunctionPointerTypeSymbol symbol)
+            {
+                var sig = symbol.Signature;
+
+                var otherReturnType = (TypeSymbol)Visit(sig.ReturnType);
+                var translationFailed = otherReturnType is null;
+                var otherRefCustomModifiers = VisitCustomModifiers(sig.RefCustomModifiers);
+                var otherReturnTypeWithAnnotations = sig.ReturnTypeWithAnnotations.WithTypeAndModifiers(otherReturnType, VisitCustomModifiers(sig.ReturnTypeWithAnnotations.CustomModifiers));
+
+                var otherParameterTypes = ImmutableArray<TypeWithAnnotations>.Empty;
+                ImmutableArray<ImmutableArray<CustomModifier>> otherParamRefCustomModifiers = default;
+
+                if (sig.ParameterCount > 0)
+                {
+                    var otherParamsBuilder = ArrayBuilder<TypeWithAnnotations>.GetInstance(sig.ParameterCount);
+                    var otherParamRefCustomModifiersBuilder = ArrayBuilder<ImmutableArray<CustomModifier>>.GetInstance(sig.ParameterCount);
+
+                    foreach (var param in sig.Parameters)
+                    {
+                        var otherType = (TypeSymbol)Visit(param.Type);
+                        if (otherType is null)
+                        {
+                            translationFailed = true;
```

Comment:
>translationFailed = true; [](start = 28, length = 25)

Seems like you can do 
```
otherParamsBuilder.Free();
otherParamRefCustomModifiersBuilder.Free();
return null;
```

here and remove the subsequent translationFailed checks.

Reply:
I was basing it on VisitNamedType, which goes through all the type arguments and then checks for failure, but if that isn't necessary I'll make it simpler.

Reply:
I don't see why it would be necessary. I think VisitNamedType only does that because it's using SelectAsArray with lambda.

---
In reply to: [425307658](https://github.com/dotnet/roslyn/pull/44256#discussion_r425307658) [](ancestors = 425307658)